typeclasses/mixins.py

```diff
@@ -509,6 +510,9 @@ def return_appearance(self, pobject, detailed=False, format_desc=False,
                 string += "\n%s{n" % desc
         else:  # for crafted objects, respect formatting
             string += "\n%s{n" % desc
+            templates = Template.objects.from_desc(string)
```

Comment:
What we probably wanna do here is have some simple caching with a `self.ndb` value. Like:
```
if self.ndb.cached_string_after_template_replacement:
    string = self.ndb.cached_string_after_template_replacement
else:
    templates = Template.objects.from_desc(string)
    if templates.exists():
        string = replace_template_values(string, templates)
    self.ndb.cached_string_after_template_replacement = string
```

Reply:
Resolved.

Reply:
Oh, hmm, crap. The caching works, but, we're gonna have to wait until I have the template object relationship finished as well, for the cascade from Template save.

Reply:
This has been resolved.